Client/FileManagerExtension.swift

```diff
@@ -40,4 +41,50 @@ public extension FileManager {
         }
         return false
     }
+    
+    func getOrCreateDirectory(withName name: String) -> (path: String, created: Bool) {
+        guard let documentDirectory = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true).first else {
+            log.error("Can't get documents dir.")
+            return ("", false)
+        }
+        
+        let path = documentDirectory + "/" + name
```

Comment:
Manually appending this scares me a bit, wdty about:

`let url = URL(string: documentDirectory)?.appendingPathComponent(name)`

We could then use `createDirectory(at: URL)` instead too? would need to be unwrapped inside the below if:

```
if let path = url?.path, !fileExists(atPath: path) {
}
```